code/modules/client/preferences.dm

```diff
@@ -1739,16 +1719,22 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 					if(unlock_content)
 						toggles ^= PREFTOGGLE_MEMBER_PUBLIC
 				if("gender")
-					if(active_character.gender == MALE)
-						active_character.gender = FEMALE
-					else
-						active_character.gender = MALE
-					active_character.underwear = random_underwear(active_character.gender)
-					active_character.undershirt = random_undershirt(active_character.gender)
-					active_character.socks = random_socks()
-					active_character.facial_hair_style = random_facial_hair_style(active_character.gender)
-					active_character.hair_style = random_hair_style(active_character.gender)
-
+					var/list/friendlyGenders = list("Male" = "male", "Female" = "female", "Other" = "plural")
+					var/pickedGender = input(user, "Choose your gender.", "Character Preference", active_character.gender) as null|anything in friendlyGenders
+					if(pickedGender && friendlyGenders[pickedGender] != active_character.gender)
+						switch(friendlyGenders[pickedGender])
+							if("plural")
+								active_character.features["body_model"] = pick(MALE, FEMALE)
+							else
+								active_character.features["body_model"] = friendlyGenders[pickedGender]
+						active_character.gender = friendlyGenders[pickedGender]
+						active_character.underwear = random_underwear(active_character.gender)
+						active_character.undershirt = random_undershirt(active_character.gender)
+						active_character.socks = random_socks()
+						active_character.facial_hair_style = random_facial_hair_style(active_character.gender)
```

Comment:
is randomizing all this necessary now that it's all generic? (the options exist on both)

Reply:
I’ll see what happens if I remove it as I’ve heard some complaining downstream that the constant randomization whenever you make a major change to a character is very annoying.

Reply:
I've removed the randomizaton that happens to these things.